fdb-record-layer-core/src/main/proto/record_metadata_options.proto

```diff
@@ -39,8 +39,9 @@ message RecordTypeOptions {
     RECORD = 1;
     UNION = 2;
     NESTED = 3;
+    UNSET = 4;
```

Comment:
Sorry, this occurred to me later. I think if `UNSET` is set to `0` (which is legal for enum ordinals but not record field ordinals), then this behaves the same in proto2 and proto3 insofar as the default value is ordinal 0, and then setting it explicitly to 0 is the same as leaving it unset (which are exactly the semantics that proto3 expects). It might be nice to adopt the proto3 idiom while we're here.

Reply:
Yeah, that had been my suggestion. Looks good now.